.github/workflows/publish-sdkman.yml

```diff
@@ -0,0 +1,69 @@
+###################################################################################################
+###                   THIS IS A REUSABLE WORKFLOW TO PUBLISH SCALA TO SDKMAN!                   ###
+### HOW TO USE:                                                                                 ###
+###   - THE RELEASE WORKFLOW SHOULD CALL THIS WORKFLOW                                          ###
+###   - IT WILL PUBLISH TO SDKMAN! THE BINARIES TO EACH SUPPORTED PLATFORM AND A UNIVERSAL JAR  ###
+###   - IT CHANGES THE DEFAULT VERSION IN SDKMAN!                                               ###
+###                                                                                             ###
+### NOTE:                                                                                       ###
+###   - WE SHOULD KEEP IN SYNC THE NAME OF THE ARCHIVES WITH THE ACTUAL BUILD                   ###
+###   - WE SHOULD KEEP IN SYNC THE URL OF THE RELEASE                                           ###
+###################################################################################################
+
+
+name: Publish Scala to SDKMAN!
+run-name: Publish Scala ${{ inputs.version }} to SDKMAN!
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+    secrets:
+      CONSUMER-KEY:
+        required: true
+      CONSUMER-TOKEN:
+        required: true 
+
+env:
+  RELEASE-URL: 'https://github.com/scala/scala3/releases/download/${{ inputs.version }}'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - platform: LINUX_64
+            archive : 'scala3-${{ inputs.version }}-x86_64-pc-linux.tar.gz'
+          - platform: LINUX_ARM64
+            archive : 'scala3-${{ inputs.version }}-aarch64-pc-linux.tar.gz'
+          - platform: MAC_OSX
+            archive : 'scala3-${{ inputs.version }}-x86_64-apple-darwin.tar.gz'
+          - platform: MAC_ARM64
+            archive : 'scala3-${{ inputs.version }}-aarch64-apple-darwin.tar.gz'
+          - platform: WINDOWS_64
+            archive : 'scala3-${{ inputs.version }}-x86_64-pc-win32.tar.gz'
+          - platform: UNIVERSAL
+            archive : 'scala3-${{ inputs.version }}.zip'
+    steps:
+      - uses: hamzaremmal/sdkman-release-action@main # TODO: Make a release of the action and configure the version here
+        with:
+          CONSUMER-KEY   : ${{ secrets.CONSUMER-KEY }}
+          CONSUMER-TOKEN : ${{ secrets.CONSUMER-TOKEN }}
+          CANDIDATE      : scala
+          VERSION        : ${{ inputs.version }}
+          URL            : '${{ env.RELEASE-URL }}/${{ matrix.archive }}'
+          PLATFORM       : ${{ matrix.platform }}
+  
+  default:
+    runs-on: ubuntu-latest
+    needs: publish
+    steps:
+      - uses: hamzaremmal/sdkman-default-action@main # TODO: Make a release of the action and configure the version here
```

Comment:
Do you have any estimate of how much effort would take to make a release for these actions? I'm mostly sceptical about using a main branch directly as it's fragile to any changes. 


A tag (preferred if possible) or even a commit pointing to well known revision might be a better idea and might prevent getting any unexpected changes in these workflows when publishing.

Reply:
Not much actually, I just have to write the documentation for it and ask the author of sdkman about the body of the reponse to parse it to extract them as outputs. (See [`action.yml`](https://github.com/hamzaremmal/sdkman-release-action/blob/main/action.yml)). In all cases, this action is fully working but I don't have the time at the moment to polish these details. 

> A tag (preferred if possible) or even a commit pointing to well known revision might be a better idea and might prevent getting any unexpected changes in these workflows when publishing.

Agreed, I will update these to use the commit instead of the branch.